LIVE-1292 Cover Polkadot existential deposit edge case

alexalouit opened the pull request:
## Context (issues, jira)

https://ledgerhq.atlassian.net/browse/LIVE-1292

## Description / Usage

When trying to emptied an account, it doesn't work.
This happens when the balance is at ~1 DOT, this is a limitation of the Polkadot blockchain.
The logic of Ledger Live Common was not adapted, this PR fix this.

This PR also contains a fix on the optimistic operation which does not contain the correct type when it is a send max.

<!-- please share a sample of code that uses your new code (if features were added) -->
<!-- please document quickly what would the "user land" need to do to use the feature -->

## Expectations

- [ ] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [ ] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

<!--
If one of these can't be checked, please document it carefully (on the reason you can't check it).
NB: We want to avoid as much as possible such breaking changes to make a bump very easy.
-->

alexalouit committed:
Fix transaction mode mapping

Send max must be balances.transfer palletMethod instead of balances.transferKeepAlive

alexalouit committed:
Fix 1 DOT Polkadot limitation

On the Polkadot network, an address is only active when it holds a minimum amount, currently set at 1 DOT.
There's a limitation when you hold ~ 1 DOT.
Use recommended 1.1 DOT value.

haammar-ledger commented on src/families/polkadot/js-signOperation.ts:
```diff
@@ -52,6 +53,10 @@ const getExtra = (type: string, account: Account, transaction: Transaction) => {
       }
     : {};
 
+  if (transaction.mode == "send" && transaction.useAllAmount) {
+    extra.palletMethod = MODE_TO_PALLET_METHOD["sendMax"];
+  }
+
```
Code readability : can you please rework the `const extra = ...` definition above to merge it with this?
You can rely on if/else as you did here, I think it's way more readable than the multiple ternary operators above.

haammar-ledger reviewed (changes requested):
Please put the link to the JIRA ticket in the description of the PR + cleanup the title a bit it looks weird :smile:

alexalouit committed:
Improve code readability

Don't use ternary operator

alexalouit committed:
Split Existential Deposit and margin

Restore EXISTENTIAL_DEPOSIT, use new constant EXISTENTIAL_DEPOSIT_RECOMMENDED_MARGIN to handle recommended margin

haammar-ledger reviewed (approved):
:ok_hand: :+1:
